### DISCUSSION TITLE
Persisting Files

### DISCUSSION THREAD

**Discussion by am486dx** [openlibhums/janeway#3628] [Q&A]
Hi - huge fan of Voyager btw :)

We're hoping to deploy Kubernetes on Janeway, but Kubernetes can't mount/persist directories with application code in them like Docker does - it will simply overwrite the folder. This is only an issue for the `src/` folder, which stores both application state information (like the logos + articles) and important code for the application.

My (hopefully) quick question is, do you know which subdirectories in `/src` store application state information? If only `/files` and `/media` and `/textures` are modified, for example, Kubernetes setup should be quite easy, we can mount/persist those folders within `/src`. However, if application state information is stored in many subdirectories all over `/src`, the amount of volume mounts required would become infeasible.

Thank you!

**Comment by ajrbyers** [comment]
Hey @am486dx,

we talked about this a few times on Discord, did you make any progress on this?

**Reply by am486dx** [reply]
Andy,

Yes, I came to what I believe is an acceptable solution. I completely forgot I had even made this post. If there's something you want to know about it let me know, otherwise I'll just close the discussion.

**Reply by ajrbyers** [reply]
Would we be able to find some time to talk about it? We want to put together something for other folks who're interested in using Janeway with Docker and K8s.

**Reply by am486dx** [reply]
Sure thing! Just shoot me a message on Discord and we can set up a time. 
Also, I've made the docker repository public here: https://github.com/USMAI-Library-Consortium/janeway-k8s-docker/tree/main
